src/nats/connection.cr

```diff
@@ -371,10 +371,7 @@ module NATS
     def close
       return if @closed
       @closed = true
-      @out.synchronize do
-        flush_outbound
-        @socket.flush
-      end
+      flush_outbound
```

Comment:
Curious why you removed the synchronize scope here?

Reply:
`flush_outbound` is defined as:

```crystal
    private def flush_outbound
      @out.synchronize do
        @socket.flush
      end
    end
```

So this block was actually running the same code inside of itself:

```crystal
      @out.synchronize do
        @out.synchronize do
          @socket.flush
        end
        @socket.flush
      end
```

The other reason is that recursively locking a mutex no longer works. I actually didn't realize it used to until I saw this code, but it looks like the mutex used to yield the block immediately if the current fiber already held the lock. That now raises an exception.

Reply:
Ah, gotcha. Thanks for the catch!